File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.py
class Solution:
    def smallestRepunitDivByK(self, k: int) -> int:
        n = 0
        c = 1
        if (k % 2 == 0 or k % 5 == 0): return -1
        while True:
            n = n * 10 + 1
            n = n % k 
            if n == 0:
                return c
            c += 1

File: 1015-smallest-integer-divisible-by-k/1015-smallest-integer-divisible-by-k.cpp
class Solution {
public:
    int smallestRepunitDivByK(int k) {
        int n = 0, c = 0;
        if (k % 2 == 0 || k % 5 == 0) return -1;
        while (true) {
            n = (n * 10) + 1;
            n = n % k;
            c += 1;
            if (n == 0) return c;
        }
    }
};